src/main/java/bio/terra/SwaggerDocumentationConfig.java

```diff
@@ -1,21 +1,84 @@
 package bio.terra;
 
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
 import springfox.documentation.builders.ApiInfoBuilder;
+import springfox.documentation.builders.PathSelectors;
 import springfox.documentation.builders.RequestHandlerSelectors;
 import springfox.documentation.service.ApiInfo;
+import springfox.documentation.service.AuthorizationScope;
 import springfox.documentation.service.Contact;
+import springfox.documentation.service.GrantType;
+import springfox.documentation.service.ImplicitGrant;
+import springfox.documentation.service.LoginEndpoint;
+import springfox.documentation.service.OAuth;
+import springfox.documentation.service.SecurityReference;
 import springfox.documentation.spi.DocumentationType;
+import springfox.documentation.spi.service.contexts.SecurityContext;
 import springfox.documentation.spring.web.plugins.Docket;
+import springfox.documentation.swagger.web.ApiKeyVehicle;
+import springfox.documentation.swagger.web.SecurityConfiguration;
 
-@javax.annotation.Generated(value = "io.swagger.codegen.languages.SpringCodegen",
-                            date = "2018-11-21T14:51:14.798-05:00")
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
 
 @Configuration
+@EnableConfigurationProperties
+@ConfigurationProperties(prefix = "oauth")
 public class SwaggerDocumentationConfig {
 
+    private String schemeName;
+    private String[] scopes;
+    private String loginEndpoint;
+    private String clientId;
+    private String clientSecret;
+    private static final String UNUSED = "__unused__";
+
+    public String getSchemeName() {
+        return schemeName;
+    }
+
+    public void setSchemeName(String schemeName) {
+        this.schemeName = schemeName;
+    }
+
+    public String getLoginEndpoint() {
+        return loginEndpoint;
+    }
+
+    public void setLoginEndpoint(String loginEndpoint) {
+        this.loginEndpoint = loginEndpoint;
+    }
+
+    public String getClientId() {
+        return clientId;
+    }
+
+    public void setClientId(String clientId) {
+        this.clientId = clientId;
+    }
+
+    public String getClientSecret() {
+        return clientSecret;
+    }
+
+    public void setClientSecret(String clientSecret) {
+        this.clientSecret = clientSecret;
+    }
+
+    public String[] getScopes() {
+        return scopes.clone();
+    }
+
+    public void setScopes(String[] scopes) {
+        this.scopes = scopes.clone();
+    }
+
     ApiInfo apiInfo() {
```

Comment:
Just to clarify, adding the `@ConfigurationProperties("oauth")` provides the prefixes coming _from_ the config file, but does not affect how these beans are references from inside the code. Right?

Reply:
Right, this is the way to tell Spring to look into application.properties and match anything with the prefix `oauth.` or to match any environment variable that start with `OAUTH_`.